### PR TITLE
[risk=low][RW-7473] Sync expiring modules on AAR load

### PR DESCRIPTION
**Pull request by jmthibault79**
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/access/access-renewal.tsx [hunk 1]
```diff
@@ -178,14 +179,11 @@ const RenewalCard = withStyle(renewalStyle.card)(
 );
 
 // Page to render
-export const AccessRenewal = fp.flow(
-  withProfileErrorModal
-)((spinnerProps: WithSpinnerOverlayProps) => {
-  useEffect(() => spinnerProps.hideSpinner(), []);
```
Comment (jmthibault79) [inline ui/src/app/pages/access/access-renewal.tsx:184]: combined with the other "onMount" actions

File: ui/src/app/pages/access/access-renewal.tsx [hunk 2]
```diff
@@ -195,28 +193,32 @@ export const AccessRenewal = fp.flow(
   const [loading, setLoading] = useState(false);
   const [, navigateByUrl] = useNavigation();
 
+  const expirableModules = modules.filter(moduleStatus => accessRenewalModules.includes(moduleStatus.moduleName));
+
+  const completeOrBypassed = moduleName => {
+    const status = modules.find(m => m.moduleName === moduleName);
+    const wasBypassed = !!status.bypassEpochMillis;
+    return wasBypassed || !isExpiring(status.expirationEpochMillis);
```
Comment (calbach) [inline ui/src/app/pages/access/access-renewal.tsx:201]: Existing code, but is something here perhaps misnamed? At a glance, this doesn't seem to correctly account for incomplete modules. Or - does complete here refer to "renewal complete", and we assume that all modules are complete b/c we're on the AAR page?

File: ui/src/app/pages/access/access-renewal.tsx [hunk 3]
```diff
@@ -195,28 +193,32 @@ export const AccessRenewal = fp.flow(
...
       setLoading(true);
       await reloadProfile();
       setLoading(false);
     };
+    const expiringModules = expirableModules
+      .filter(status => status.expirationEpochMillis && isExpiring(status.expirationEpochMillis))
+      .map(status => status.moduleName);
 
+    externalSyncModules(expiringModules);
     getProfile();
```
Comment (calbach) [inline ui/src/app/pages/access/access-renewal.tsx:218]: Existing issue, but these two calls are async, yet we immediately call `hideSpinner`. Is that working as intended? Is this handled in a different way on the page already?

File: ui/src/app/utils/access-utils.tsx [hunk 4]
```diff
@@ -309,3 +305,13 @@ export const eligibleForTier = (profile: Profile, accessTierShortName: string):
   const rtEligiblity = profile.tierEligibilities.find(t => t.accessTierShortName === accessTierShortName)
   return rtEligiblity?.eligible
 };
+
+export const externalSyncModules = (moduleNames: AccessModule[]) => {
```
Comment (calbach) [inline ui/src/app/utils/access-utils.tsx:309]: optional nit: this reads like a noun currently, maybe `syncModulesExternal` or `syncModulesExternally`

**Review by calbach — APPROVED** [review]

File: ui/src/app/pages/access/access-renewal.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/access/access-renewal.tsx]: yeah... this code needs some extra care.  I haven't quite wrapped my head around it, but I'm planning to come back to it soon.

File: ui/src/app/utils/access-utils.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/utils/access-utils.tsx]: ok

File: ui/src/app/pages/access/access-renewal.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/access/access-renewal.tsx]: updated this logic (here and DAR) ... I'll test locally again as well

Comment (jmthibault79) [reply ui/src/app/pages/access/access-renewal.tsx]: The network timing diagram shows that we now wait for sync-training before refreshing the profile.

**Review by calbach — APPROVED** [review]

File: ui/src/app/pages/access/access-renewal.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/access/access-renewal.tsx]: PR #6089 is intended to address this

**Pull request merged**